### PR TITLE
TOML: unify toml key parsing

### PR DESCRIPTION
**Pull request by Undin**
Previously, we parsed [dotted keys](https://toml.io/en/v1.0.0-rc.3#keys) in `key1.key2=value` and `[key1.key2]` differently that was quite confusing. Now they are parsed in the same way

changelog: unify toml key parsing

### CODE REVIEW

File: intellij-toml/src/main/grammars/TomlParser.bnf [hunk 1]
```diff
@@ -42,16 +42,14 @@ private FileForm_recover ::=
...
 NewLineKeyValue ::= <<atNewLine KeyValue>> {
     elementType = KeyValue
     recoverWhile = EatUntilNextLine_recover
 }
 
-fake Key ::= BasicKey ('.' BasicKey)*
-LongKey ::= BasicKey ('.' BasicKey)* { elementType = Key }
-ShortKey ::= BasicKey { elementType = Key }
-private BasicKey ::= BareKey | BASIC_STRING | LITERAL_STRING
+Key ::= SimpleKey ('.' SimpleKey)*
```
Comment (vlad20012) [inline intellij-toml/src/main/grammars/TomlParser.bnf:51]: I don't like `SimpleKey` name. Maybe `KeySegment`?
Also, there is an option to parse keys with left recursion. We parser Rust paths this way. In this case, SimpleKey will not be needed. Pros/cons?

Comment (Undin) [reply intellij-toml/src/main/grammars/TomlParser.bnf]: Discussed in person. As a result:
- rename `SimpleKey` to `KeySegment`
- we currently don't know the actual cons of "parse keys with left recursion". Also, original TOML grammar parse keys also as segments list so we decided to keep current implementation

**Comment by Undin** [comment]
bors retry

**Comment by Undin** [comment]
bors r=vlad20012

**Pull request merged**